server/container_portforward.go

```diff
@@ -25,17 +25,19 @@ func (s StreamService) PortForward(ctx context.Context, podSandboxID string, por
 	ctx, span := log.StartSpan(ctx)
 	defer span.End()
 
-	// if we error in this function before Copying all of the content out of the stream,
-	// this stream will eventually get full, which causes leakages and can eventually brick CRI-O
+	// Drain the stream to prevent failure to close the connection and memory leakage.
 	// ref https://bugzilla.redhat.com/show_bug.cgi?id=1798193
-	emptyStreamOnError := true
+	// ref https://issues.redhat.com/browse/OCPBUGS-30978
 	defer func() {
-		if emptyStreamOnError && stream != nil {
-			go func() {
-				_, copyError := pools.Copy(io.Discard, stream)
-				log.Errorf(ctx, "Error closing port forward stream after other error: %v", copyError)
-			}()
+		if stream == nil {
+			return
 		}
```

Comment:
If we keep the if statement, but remote the `emptyStreamOnError` from it, then we have the smallest changeset.

Reply:
I like the updated structure

Reply:
If there is no objections about it, I'll keep the update one. thanks!